### ISSUE TITLE
[Bug]: Cannot store file ... is not an object of type "meta"

### ISSUE THREAD

**Issue by raintonr** [iobroker-community-adapters/ioBroker.chromecast#323]
### Contact Details

robin@rainton.com

### What happened?

Since JS controller 6.x (or maybe before, but this is when I noticed) trying to play local files results in an error.

### Adapter Version

3.3.0

### JS-Controller Version

6.0.11

### Node JS Version

v20.17.0

### NPM Version

10.8.2

### Operating System

Ubuntu

### Relevant log output

```shell
2024-10-28 16:14:15.473  - error: chromecast.0 (1938046) Cannot write file chromecast.0.files/<device ID>/exportedMedia.mp3: chromecast.0 is not an object of type "meta"
2024-10-28 16:14:15.474  - error: chromecast.0 (1938046) <device> - Cannot store file "<filename>.mp3" into chromecast.0.files/<device ID>/exportedMedia.mp3: Error: chromecast.0 is not an object of type "meta"
```

**Comment by mcm1957** [comment]
Lets add the reason already reported by @Apollon77 

https://t.me/ioBroker_dev_en/1240
`
yes seems that the "files" object is not created at all ... so an entry in io-package like https://github.com/ioBroker/ioBroker.sayit/blob/master/io-package.json#L226-L234 is needed for "files"
`

**Comment by raintonr** [comment]
I think we should just remove the feature to copy local files so they can be served by webserver, and make it mandatory that the local user makes any files they want to play available on their network beforehand. This is easy enough with the webserver adapter anyway and more efficient.

**Comment by Thobe-4** [comment]
Sorry for the maybe stupid question: What do you mean with "make it mandatory that the local user makes any files they want to play available on their network beforehand"?
I have a mp3 file stored locally (i.e. on the raspberry the iobroker runs on, in the /home/iobroker folder. I get the same errors mentioned above when trying to play the mp3 via url2play. What exectly do I need to setup to be (again) able to play the mp3 file (it worked until I recently updated the js-controller.
Many thanks!

**Comment by raintonr** [comment]
Use a webserver on your LAN to make the file accessible on some URL via http(s). You can use the webserver adapter for that or a standalone daemon. For example I use nginx on a different host.

Then, instead of using `/my/local/file.mp3` specify `http://my.local.webserver/whatever/you/setup.mp3` in `url2play`.

**Comment by mcm1957** [comment]
> Sorry for the maybe stupid question: What do you mean with "make it mandatory that the local user makes any files they want to play available on their network beforehand"? I have a mp3 file stored locally (i.e. on the raspberry the iobroker runs on, in the /home/iobroker folder. I get the same errors mentioned above when trying to play the mp3 via url2play. What exectly do I need to setup to be (again) able to play the mp3 file (it worked until I recently updated the js-controller. Many thanks!

If there is no technical reason to block local files, I would suggest to keep the feature to acces local files and solve the problem. 

@Thobe-4 
Please add a log of the instance with loglevel set to debug. MAybe one can see in more detail where the problem arises.

**Comment by raintonr** [comment]
Well, having the Chromecast adapter copy the file every time is inefficient. 

Plus, code change is harder than code removal 😉

**Comment by Thobe-4** [comment]
Thanks for the quick responses. I will try to setup a webserver and report back if it works.

When trying to open a local file, there are two errors in the log:


chromecast.0 | 2025-01-27 20:30:07.988 | error | Esszimmer - Cannot store file "/home/iobroker/mp3/[Filename].mp3" into chromecast.0.files/[Device ID]/exportedMedia.mp3: Error: chromecast.0 is not an object of type "meta"

chromecast.0 | 2025-01-27 20:30:07.987 | error | Cannot write file chromecast.0.files/[Device ID]/exportedMedia.mp3: chromecast.0 is not an object of type "meta"

Hope this helps!

**Comment by mcm1957** [comment]
Thanks for info

So I will close her as the initial problem seems to be solved.

**Comment by raintonr** [comment]
> So I will close her as the initial problem seems to be solved.

Not really. Uploading to a local webserver is the solution I proposed over a year ago. If a user tries to put a local path in the url2play object it will still fail. 

I think at least we are agreed making files available via a webserver is the answer now so should remove that code that attempts to copy local files.

**Comment by raintonr** [comment]
> If I enter this URL into url2play, the sound will play:\n\nhttp://192.168.0.222:8082/0_userdata.0/1.mp3

Yes, of course. Because that's a HTTP URL and not a local file. 

If you are not putting a local filename in the url2play object this issue will not affect you.
